website/addons/googledrive/model.py

```diff
@@ -111,218 +101,183 @@ def get_or_create(cls, node, path):
 
-    @username.setter
-    def username(self, val):
-        self.oauth_settings.username = val
+    _api = None
 
     @property
-    def access_token(self):
-        if self.oauth_settings:
-            return self.oauth_settings.access_token
-        return None
-
-    @access_token.setter
-    def access_token(self, val):
-        self.oauth_settings.access_token = val
+    def api(self):
+        """Authenticated ExternalProvider instance"""
+        if self._api is None:
+            self._api = GoogleDriveProvider()
+            self._api.account = self.external_account
+        return self._api
 
     @property
-    def refresh_token(self):
-        if self.oauth_settings:
-            return self.oauth_settings.refresh_token
-        return None
-
-    @refresh_token.setter
-    def refresh_token(self, val):
-        self.oauth_settings.refresh_token = val
-
-    @property
-    def expires_at(self):
-        if self.oauth_settings:
-            return self.oauth_settings.expires_at
-        return None
+    def complete(self):
+        return bool(self.has_auth and self.user_settings.verify_oauth_access(
+            node=self.owner,
+            external_account=self.external_account,
+            metadata={'folder': self.drive_folder_id}
+        ))
 
-    @expires_at.setter
-    def expires_at(self, val):
-        self.oauth_settings.expires_at = val
+    def set_folder(self, folder, auth, add_log=True):
+        self.drive_folder_id = folder['id']
+        self.folder_path = folder['path']
 
     @property
-    def has_auth(self):
-        if self.oauth_settings:
-            return self.oauth_settings.access_token is not None
-        return False
-
-    def fetch_access_token(self, force_refresh=False):
-        if self.oauth_settings:
-            return self.oauth_settings.fetch_access_token(force_refresh)
-        return None
-
-    def clear(self):
-        if self.oauth_settings:
-            self.oauth_settings.revoke_access_token()
-            self.oauth_settings = None
-            self.save()
-
-        for node_settings in self.googledrivenodesettings__authorized:
-            node_settings.deauthorize(Auth(self.owner))
-            node_settings.save()
-
-    def save(self, *args, **kwargs):
-        if self.oauth_settings:
-            self.oauth_settings.save()
-        return super(GoogleDriveUserSettings, self).save(*args, **kwargs)
-
-    def delete(self, save=True):
-        self.clear()
-        super(GoogleDriveUserSettings, self).delete(save)
-
-    def __repr__(self):
-        return u'<GoogleDriveUserSettings(user={self.owner.username!r})>'.format(self=self)
-
-
-class GoogleDriveNodeSettings(StorageAddonBase, AddonNodeSettingsBase):
-
-    folder_id = fields.StringField(default=None)
-    folder_path = fields.StringField(default=None)
-
-    user_settings = fields.ForeignField(
-        'googledriveusersettings', backref='authorized'
-    )
+    def provider_name(self):
```

Comment:
Probably because Zendeley has it being a property, which isn't a good reason. Changing